cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/datastore/SavedAttachment.java

```diff
@@ -57,7 +57,7 @@
     private final File file;
     private final AttachmentStreamFactory attachmentStreamFactory;
 
-    protected SavedAttachment(long seq, String name, byte[] key, String type, Encoding encoding,
+    public SavedAttachment(long seq, String name, byte[] key, String type, Encoding encoding,
```

Comment:
Agreed. `AttachmentsForRevisionCallable` has been migrated partly because it was needed for other callables - `GetAllRevisionsOfDocumentCallable` and `GetDocumentCallable`. But I'm going to hold off on migrating any of the other `AttachmentManager` stuff because it will make this PR too big.